src/modules/dashboard/components/ActionsPage/ActionsPage.tsx

```diff
@@ -0,0 +1,289 @@
+             */
+            <span className={styles.addressInTitle}>
+              <CopyableAddress>{walletAddress}</CopyableAddress>
+            </span>
+          );
+        }
+        return false;
+      })(),
+      name: (() => {
+        if (transactionData?.transaction?.events?.length) {
+          const { events } = transactionData.transaction;
+          /*
+           * Display the first event as the page title
+           * We might need to change this in the future
+           */
+          return events[0].name;
+        }
+        return false;
+      })(),
+    }),
+    [transactionData, fallbackUserData, userData],
+  );
+
+  if (!isTransactionFormat(transactionHash) || transactionDataError) {
+    return (
+      <div className={styles.main}>
+        <div className={styles.notFoundContainer}>
+          <NakedMoleImage />
+          <Heading
+            text={MSG.transactionNotFound}
+            appearance={{
+              size: 'medium',
+              weight: 'medium',
+              theme: 'dark',
+            }}
+          />
+          <Button
+            title={MSG.returnToColony}
+            text={MSG.returnToColony}
+            linkTo={`/colony/${colonyName}`}
+            appearance={{
+              theme: 'primary',
+              size: 'large',
+            }}
+          />
+          <div className={styles.divider} />
+          <div className={styles.hashWrapper}>
+            <Hash transactionHash={transactionHash || ''} />
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  if (transactionDataLoading || userDataLoading || !transactionData) {
+    return <LoadingTemplate loadingText={MSG.loading} />;
+  }
+
+  if (
+    !colonyName ||
+    (reverseENSAddress as SuperSpecificColonyAddress) instanceof Error
+  ) {
+    return <Redirect to={NOT_FOUND_ROUTE} />;
+  }
+
+  const {
+    transaction: { hash, status, events },
+  } = transactionData;
+
+  return (
+    <div className={styles.main}>
+      <div className={styles.container}>
+        <div className={styles.content}>
+          {/*
+           * @NOTE Can't use `Heading` here since it uses `formmatedMessage` internally
+           * for message descriptors, and that doesn't support our complex text values
+           */}
+          <h1 className={styles.heading}>
+            <FormattedMessage
+              {...MSG.actionTitle}
+              values={titleDynamicValues}
+            />
+          </h1>
+          {!events && hash && (
+            <TransactionHash
+              transactionHash={hash}
+              /*
+               * @NOTE Otherwise it interprets 0 as false, rather then a index
+               * Typecasting it doesn't work as well
+               */
+              status={typeof status === 'number' && STATUS_MAP[status]}
+            />
+          )}
+          <ul>
+            <b>Events for the tx:</b>
+            {events.map(({ name, topic }) => (
+              <li key={topic || ''}>
```

Comment:
empty string as a key? i think it will throw console warning if we won't have topic

Reply:
I don't care about this, this mostly debug log. 

This will be replaced by the components you are building in DEV-47